### PR TITLE
dht: fixing mask length and offset

### PR DESCRIPTION
**Pull request by gjerecze**

### CODE REVIEW

File: lib/dht/DhtClient.cpp [hunk 1]
```diff
@@ -149,11 +148,11 @@ void DhtClient::initialize(DhtCore *dhtCore) {
     erpc::Rpc<erpc::CTransport> *rpc;
 
     int i = _dhtCore->numberOfClientThreads++;
-    _remoteRpcId = (dhtCore->randomSeed + i) % _numServerThreads;
+    _remoteRpcId = (dhtCore->randomSeed + i) % dhtCore->getDhtThreadsCount();
```
Comment (aljer) [inline lib/dht/DhtClient.cpp]: When work in local mode with --n-dht-threads=0 causes 'floating point exception'

Comment (gjerecze) [reply lib/dht/DhtClient.cpp]: I cannot reproduce the issue. Could you provide the exact command? With --n-dht-threads=0 the DhtClient is not initialized at all.

File: .gitmodules [hunk 2]
```diff
@@ -29,4 +29,3 @@
 [submodule "third-party/eRPC-fork"]
 	path = third-party/eRPC-fork
 	url = https://github.com/daq-db/eRPC.git
-	ignore = dirty
```
Comment (jakubradtke) [inline .gitmodules]: is it intentional change?

File: apps/minidaq/minidaq.cfg.in [hunk 3]
```diff
@@ -10,18 +10,16 @@
...
+                    // Local node
+                    ip = "10.10.10.53";
+                    port = 31850;
+                    local = 1;
+                },
+                {
+                    // Storage node
+                    ip = "10.10.10.55";
+                    port = 31850;
+                    keys = { start = "101", end = "255" }
```
Comment (jakubradtke) [inline apps/minidaq/minidaq.cfg.in]: not sure why "start" is changed to "101"?

File: include/daqdb/Options.h [hunk 4]
```diff
@@ -148,32 +148,26 @@ struct RuntimeOptions {
...
 
 struct DhtNeighbor {
     std::string ip;
     unsigned short port = 0;
-    /**
-     * Second port is used for additional eRPC channel that is used for
-     * communication between storage nodes.
-     * Should be set only for local node.
-     */
-    unsigned short peerPort = 0;
```
Comment (jakubradtke) [inline include/daqdb/Options.h]: Cool, removing this option and using rpcId improves code readability a lot!

File: examples/clinode/clinode.cfg.in [hunk 5]
```diff
@@ -44,23 +44,20 @@ runtime_base_core_id = 0;
...
+neighbors : (
+                {
+                    // replace with ip of storage node
+                    ip = "10.10.10.54";
+                    port = 31850;
+                    local = 1;
+                    keys = { start = "0", end = "64" }
+                },
+                {
+                    // replace with ip of peer storage node
```
Comment (jakubradtke) [inline examples/clinode/clinode.cfg.in]: maybe would be good to comment peer node section?
clinode is failing due to that option if not correct IP is set.

File: .gitmodules (hunk 2 above)
Comment (gjerecze) [reply .gitmodules]: accidental, fixed

File: apps/minidaq/minidaq.cfg.in (hunk 3 above)
Comment (gjerecze) [reply apps/minidaq/minidaq.cfg.in]: fixed

File: examples/clinode/clinode.cfg.in (hunk 5 above)
Comment (gjerecze) [reply examples/clinode/clinode.cfg.in]: I'm not sure what you mean. Extending the doc here or the main readme?

File: include/daqdb/Options.h (hunk 4 above)
Comment (gjerecze) [reply include/daqdb/Options.h]: Indeed ;)

File: lib/dht/DhtClient.cpp (hunk 1 above)
Comment (gjerecze) [reply lib/dht/DhtClient.cpp]: Issue resolved offline - configuration was wrong

**Pull request merged**